### ISSUE TITLE
twisted.python.log.DefaultObserver ignores `why' argument of event dictionary

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1982]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1982|
|Type|defect|
|Created|2006-08-02 13:36:37Z|
|Branch|https://github.com/twisted/twisted/tree/defaultObserver-why-1982|

This is somewhat confusing, since it changes what information is available for debugging errors based on whether DefaultObserver or FileLogObserver is being used.

Attachments:

* [1982.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/e18/e182c2172761f9deac3cdc797925b0b32547a1c1/26da14f6485bc06c16c1d22f21708ae3203ccfe7.patch) (1938 bytes) - added by aruse on 2011-03-14 15:16:07Z - Patch to resolve #1982

<details><summary>Searchable metadata</summary>

```
trac-id__1982 1982
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_defaultObserver_why_1982 branches/defaultObserver-why-1982
branch_author__cyli cyli
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1154525797000000 1154525797000000
changetime__1474197555542286 1474197555542286
version__None None
owner__ 
cc__aruse
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="aruse's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|aruse set owner to aruse<br>aruse set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="aruse's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|aruse removed owner<br>aruse set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_6"></a>|@cyli commented|
|-|-|

(In [31021]) Branching to 'defaultObserver-why-1982'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_7"></a>|@cyli commented|
|-|-|

(In [31024]) Applying aruse's patch

refs [#1982](https://github.com/twisted/twisted/issues/1982)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_8"></a>|@cyli commented|
|-|-|

[Tests seem to pass](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/defaultObserver-why-1982), except for an unrelated epoll bug.  Merging.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_9"></a>|@cyli commented|
|-|-|

Thanks for working on this aruse!

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_10"></a>|@cyli commented|
|-|-|

(In [31030]) Added news file

refs [#1982](https://github.com/twisted/twisted/issues/1982)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_11"></a>|@cyli set status to `closed`|
|-|-|

(In [31031]) Merge defaultObserver-why-1982: Add 'why' argument to debugging error

Author: aruse
Reviewer: cyli
Fixes: [#1982](https://github.com/twisted/twisted/issues/1982)

Add 'why' argument of event dictionary to stderr output produced by twisted.python.log.DefaultObserver

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_12"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
